parametric_stellarator.py

```diff
@@ -380,45 +512,80 @@ def parametric_stellarator(
+                'step_export': export component STEP files (bool, defaults to
+                    True),
+                'h5m_export': export DAGMC-compatible neutronics H5M file using
+                    Cubit or Gmsh. Acceptable values are None or a string value
+                    of 'Cubit' or 'Gmsh' (str, defaults to None). The string is
+                    case-sensitive. Note that if magnets are included, 'Cubit'
+                    must be used,
+                'plas_h5m_tag': optional alternate material tag to use for
+                    plasma. If none is supplied and the plasma is not excluded,
+                    'plasma' will be used (str, defaults to None),
+                'facet_tol': maximum distance a facet may be from surface of
+                    CAD representation for Cubit export (float, defaults to
+                    None),
+                'len_tol': maximum length of facet edge for Cubit export
+                    (float, defaults to None),
+                'norm_tol': maximum change in angle between normal vector of
+                    adjacent facets (float, defaults to None),
+                'min_mesh_size': minimum mesh element size for Gmsh export
+                    (float, defaults to 5.0),
+                'max_mesh_size': maximum mesh element size for Gmsh export
+                    (float, defaults to 20.0),
+                'volume_atol': absolute volume tolerance to allow when matching
+                    parts in intermediate BREP file with CadQuery parts for
+                    Gmsh export(float, defaults to 0.00001),
+                'center_atol': absolute center coordinates tolerance to allow
+                    when matching parts in intermediate BREP file with CadQuery
+                    parts for Gmsh export (float, defaults to 0.00001),
+                'bounding_box_atol': absolute bounding box tolerance  to allow
+                    when matching parts in intermediate BREP file with CadQuery
+                    parts for Gmsh export (float, defaults to 0.00001).
+            }
+        logger (object): logger object (defaults to None). If no logger is
+            supplied, a default logger will be instantiated.
     """
     # Check if total toroidal extent exceeds 360 degrees
-    if gen_periods > num_periods:
-        raise ValueError(
-            'Number of requested periods to generate exceeds number in '
+    try:
+        assert gen_periods <= num_periods, \
+            'Number of requested periods to generate exceeds number in ' \
             'stellarator geometry'
-        )
-
-    # Check that h5m_export has an appropriate value
-    if h5m_export != None and h5m_export != 'Cubit' and h5m_export != 'Gmsh':
-        raise ValueError(
-            'h5m_export must be None or have a string value of \'Cubit\' or '
-            '\'Gmsh\''
-        )
+    except AssertionError as e:
+        logger.error(e.args[0])
+        raise e
```

Comment:
Note here that I used a different implementation for exception handling. Because there is no script level outside this function aside from the calling script (this is the main function), I couldn't put a try statement on the outside of this functionality. I didn't want to throw this check inside a different function (such as `stellarator_torus`) because I figured it didn't make sense to check it multiple times (as `stellarator_torus` gets called multiple times)

Reply:
OK - but I think you need to initialize the logger first - see a few lines below...